ethcore/private-tx/src/messages.rs

```diff
@@ -28,12 +28,28 @@ pub struct PrivateTransaction {
 	pub encrypted: Bytes,
 	/// Address of the contract
 	pub contract: Address,
+	/// Hash
+	hash: H256,
 }
 
 impl PrivateTransaction {
-	/// Compute hash on private transaction
+	/// Constructor
+	pub fn new(encrypted: Bytes, contract: Address) -> Self {
+		PrivateTransaction {
+			encrypted,
+			contract,
+			hash: 0.into(),
+		}.compute_hash()
```

Comment:
please get rid of the `compute_hash` function and just compute hash inline

Reply:
IMO such optimization in order to calculate hash only once is justified. I've reworked it in order to provide access to the fields via getters